addons/website/static/src/snippets/s_website_form/options.js

```diff
@@ -955,6 +956,9 @@ options.registry.WebsiteFieldEditor = FieldEditor.extend({
     init: function () {
         this._super.apply(this, arguments);
         this.rerender = true;
+        this._getVisibilityConditionCachedRecords = memoize((...args) => {
+            return this.orm.searchRead(...args);
```

Comment:
This will die if used on a huge table. Maybe force a limit parameter to prevent the dropdown from ending up with millions of product names.